### ISSUE TITLE
False positive for "Use of moved value" when using `impl AsRef<T> + Copy`

### ISSUE THREAD

**Issue by Alphare** [intellij-rust/intellij-rust#5410]
## Environment

* **IntelliJ Rust plugin version: 0.2.122.3105-201**
* **Rust toolchain version: N/A**
* **IDE name and version: PyCharm 2020.1 Professional**
* **Operating system: Linux**

## Problem description
Functions passing directly an `impl AsRef<T> + Copy` parameter to another function with an `AsRef<T> + Copy` parameter lead to a false positive `use of moved value`.

## Steps to reproduce
```rust
use std::path::Path;

fn bar(a: impl AsRef<Path> + Copy) {
    println!("{}", a.as_ref().display());
}

fn baz(a: impl AsRef<Path> + Copy) {
    println!("{}", a.as_ref().display());
}

fn foo(a: impl AsRef<Path> + Copy) {
    bar(a);
    baz(a);  // Use of moved value, yet compiles
}
```

**Comment by Uriopass** [comment]
Works on my machine
![image](https://user-images.githubusercontent.com/5420739/82539815-5ab83f00-9b4e-11ea-9caa-de24dfb5f90f.png)

## Environment
IntelliJ Rust plugin version: 0.2.122.3105-201
Rust toolchain version: 1.42
IDE name and version: CLion 2020.1
Operating system: Windows 10

**Comment by Undin** [comment]
Works for me with `0.4.170.4627-221`, so let's close it
@Alphare if you still have this issue, don't hesitate to reopen it
